modules/aruco/tutorials/charuco_detection/charuco_detection.markdown

```diff
@@ -102,29 +113,17 @@ ChArUco corners are interpolated from markers.
 
 The function that detect the ChArUco corners is ```cv::aruco::interpolateCornersCharuco()``` . This example shows the whole process. First, markers are detected, and then the ChArUco corners are interpolated from these markers.
 
-@code{.cpp}
-    cv::Mat inputImage;
-    cv::Mat cameraMatrix, distCoeffs;
-    // camera parameters are read from somewhere
-    readCameraParameters(cameraMatrix, distCoeffs);
-    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
-    cv::Ptr<cv::aruco::CharucoBoard> board = cv::aruco::CharucoBoard::create(5, 7, 0.04, 0.02, dictionary);
-    ...
-    std::vector<int> markerIds;
-    std::vector<std::vector<cv::Point2f>> markerCorners;
-    cv::aruco::detectMarkers(inputImage, board.dictionary, markerCorners, markerIds);
-
-    // if at least one marker detected
-    if(markerIds.size() > 0) {
-        std::vector<cv::Point2f> charucoCorners;
-        std::vector<int> charucoIds;
-        cv::aruco::interpolateCornersCharuco(markerCorners, markerIds, inputImage, board, charucoCorners, charucoIds, cameraMatrix, distCoeffs);
-    }
-@endcode
+@snippet samples/tutorial_charuco_create_detect.cpp inputImg
+@snippet samples/tutorial_charuco_create_detect.cpp matdiscoff
+@snippet samples/tutorial_charuco_create_detect.cpp dictboard
+@snippet samples/tutorial_charuco_create_detect.cpp midcornerdet
+@snippet samples/tutorial_charuco_create_detect.cpp msizecond
+@snippet samples/tutorial_charuco_create_detect.cpp charidcor
+@snippet samples/tutorial_charuco_create_detect.cpp condclose
```

Comment:
Series of snippets doesn't looks good. Please use single large in tutorial.

Reply:
For that I have to include the codes with @code tags because the documentation is designed based on the previous code snippets and the series of explanation in the tutorial is according to the codes. In the issues @asmorkalov had mentioned the use of snippets from the cpp file. Either I have to use the @code tag to manage the tutorial as designed by the author or  I have to reformat the tutorial in a different manner. What would be better to do?

Reply:
> use of snippets from the cpp file

That's correct. But there are 7 snippets in a row with 1-2 code lines. Should be one.
You can duplicate code of "inputImg" in separate functions if you need that.
Please refer on existed practice in OpenCV project.

Reply:
I have reformatted the documentation according to the existing tutorial documents. Also recently I was exploring the qrdetection library and found that there was no sample file for qrdetection in python, so I have written one. Can I make a PR for that in the opencv 3.4 branch?

Reply:
> qr

Please take a look on existed C++ samples as a reference too.

Reply:
Sure.